Assets/Scripts/Framework/Train/TrainController.cs

```diff
@@ -118,15 +123,20 @@ private void OnTick()
         [ObserversRpc(RunLocally = true)]
         private void MoveTrain(float distanceAlongSpline)
         {
+            Vector3 positionSum = Vector3.zero;
             UpdateWagonPosition(_frontWagon, distanceAlongSpline);
 
+            positionSum += _frontWagon.position;
+
             int wagons = _followingWagons.Length;
```

Comment:
```suggestion
            int wagonsAmount = _followingWagons.Length;
```
I name these in a way that you can instantly link them to the for loop. This is just a suggestion and not a must.